internal/worker/caasoperatorprovisioner/worker.go

```diff
@@ -298,6 +298,10 @@ func (p *provisioner) updateOperatorConfig(appName, password string, prevCfg caa
 			return nil, errors.NotSupportedf("operator storage provider %q", spType)
 		}
 	}
+
+	info.ImageDetails.RegistryPath = prevCfg.ImageDetails.RegistryPath
+	info.BaseImageDetails.RegistryPath = prevCfg.ImageDetails.RegistryPath
```

Comment:
This won't work in all cases. `prevCfg` might be empty.
```
		var prevCfg caas.OperatorConfig
		if op != nil && op.Config != nil {
			prevCfg = *op.Config
		}
```
This is also reinforced by the deletion of this line in the test
`c.Assert(config.ImageDetails.RegistryPath, gc.Equals, "juju-operator-image")`
since `RegistryPath` is ""

We should get the required details from the model operator like we do elsewhere.

Reply:
hi ian, I realised we could just use the info as it is since we shifted the logic to generate the registry path from the model deployment image to the server side, and we are calling OperatorProvisioningInfo through the facade as well, so we don't have to use prevCfg for the registryPath and could just use it directly from the info provided by the OperatorProvisioningInfo api. Please let me know if there's any concerns doing so :)